### ISSUE TITLE
Make NPCs names hide when behind blocks (like when sneaking) without sneaking

### ISSUE THREAD

**Issue by madtomic** [CitizensDev/Citizens2#1676]
How can I make citizens names hidden if the NPC is behind a block?

**Comment by mcmonkey4eva** [comment]
`/npc sneak` if you have Denizen installed ( https://www.spigotmc.org/resources/denizen.21039/ ) per the standard Minecraft functionality on that.

Not sure offhand if there's a way to make names hide like sneaking without actually making the NPC sneak.

**Comment by madtomic** [comment]
I will try this Denizen. Thanks for getting back. I thought about using /npc name but it just a toggle to turn the nameplate on and off.

**Comment by madtomic** [comment]
Well, sneak is is not what I am looking for. What I need is to hide the NPC names if they are no in line of sight with a player. So if a NPC is behind a wall of stones, their nameplate would be hidden. When the player come in line of sight with the NPC. NPC nameplate would be visual relative to the player looking at it.

**Comment by madtomic** [comment]
I wonder if /npc name can be schedule to toggle based on line of sight with NPC and player?

**Comment by madtomic** [comment]
Yes but if the NPC isn't sneaking. Their nameplates can be seen through walls.

![image](https://user-images.githubusercontent.com/6004542/51567878-500c6d80-1e66-11e9-8dcf-655fab7f11b7.png)

**Comment by madtomic** [comment]
As you can see there are others NPC nameplates in the background further from where I am looking at. It's annoying.

**Comment by madtomic** [comment]
Is there anything can be done to achieve this? Maybe using ProtocolLib?

**Comment by mcmonkey4eva** [comment]
I'm going to change this to a feature request for a way to make NPC nameplates hide (as though they were sneaking, but without actually sneaking)

**Comment by madtomic** [comment]
Thank you and I really hope this feature will get added in future release.

**Comment by axemwa** [comment]
this would be a cool feature to add

**Comment by madtomic** [comment]
So did you come up with something after a year of research?

**Comment by mcmonkey4eva** [comment]
You could theoretically write a Denizen script that dynamically uses hide_entity/show_entity with holograms to achieve this

**Comment by madtomic** [comment]
Any chance you could write one?

**Comment by mylo081** [comment]
Alright I know I'm late, but here is my solution. make your npc stand and make a denizen script for whatever, it just has to have one line as the first action run from a trigger:
- animate <npc> animation:sneak
hope i could help!
(p.s. no this wont make them sneak, it will just give their nameplate the effect)

**Comment by Alejandro-Sosa** [comment]
Any updates on this? Now that citizens has holograms would be a cool functionality aswell. Its easier to do it with holograms then nameplates, perhaps make the nameplate a hologram then aswell.

**Comment by mcmonkey4eva** [comment]
This is really late as I apparently forgot to post it here, but a system to hack fake-sneak into packets was added to Denizen, and there's a sample script to automatically apply it to an NPC @ https://forum.denizenscript.com/threads/tiny-sample-scripts-to-start-with.354/
